### PR TITLE
review 

### PR DESCRIPTION
**Pull request by n-poulsen**
Currently I've just reformatted the markdown (so that the maximum line length isn't too long) and fixed a typo. You can view a preview of the rendered markdown [here](https://github.com/DeepLabCutAIResidency/DLC_AI2024/blob/f9843d39c90d7490efa66d1d698ce7d6735f8f34/guides/Benchmarking_shuffle_guide.md), which does not change.

### CODE REVIEW

File: guides/Benchmarking_shuffle_guide.md [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+When benchmarking different models, maintaining the same test-train data split is 
+crucial for ensuring comparability. If the models use differing train and test datasets,
+their performance metrics cannot be accurately compared because they are not trained on
+the same data. This is especially important when comparing the performance of different
+models, such as a TensorFlow model and a PyTorch model, or two models with different
+architectures. Using the same training set is necessary to ensure fair comparisons, as
+different training sets may yield different results and make it difficult to accurately
+compare the models' performance.
```
Comment (n-poulsen) [inline guides/Benchmarking_shuffle_guide.md]: >  If the models use differing train and test datasets, their performance metrics cannot be accurately compared because they are not trained on the same data.

I'd go a bit more in depth here, especially for the fixed test set - I think we can emphasize:
- Comparing metrics on different test sets does not make sense. Take the simple case where there's a single test image. If we compare the RMSE of a model on an "easy" image against the RMSE of another model on a "hard" image, that doesn't actually tell us which model is better. We need to compare the errors the models make on the same images. 

> Using the same training set is necessary to ensure fair comparisons, as different training sets may yield different results and make it difficult to accurately compare the models' performance.

Here I would maybe emphasize more that the model performance cannot be compared: we don't know if one model is better than the other because the architecture performs better, or if it's because the images it was trained on were "easier"/"better" to learn from - we want to "decouple" the dataset from the model architecture.

File: guides/Benchmarking_shuffle_guide.md [hunk 2]
```diff
@@ -31,34 +43,57 @@ Creating a new shuffle with the same train/test split as an existing one:
...
-        - split: The data split for the shuffle. The integer value on its own does not hold any meaning, but this "split" value indicates which shuffles have the same split (as their results can then be compared)
+        - split: The data split for the shuffle. The integer value on its own does not
+hold any meaning, but this "split" value indicates which shuffles have the same split 
+(as their results can then be compared)
         - engine: Whether it's a PyTorch or TensorFlow shuffle
 ![view_existing_sh](<assets/Screenshot 2024-07-29 at 17.10.29.png>)
-5. Choose the index of the training shuffle you want to replicate. Let's assume we want to replicate the train-test split from OpenfieldOct30-trainset95shuffle3, in which split: 3 in this case, we insert in the *'From shuffle'* menu
+5. Choose the index of the training shuffle you want to replicate. Let's assume we want
+to replicate the train-test split from OpenfieldOct30-trainset95shuffle3, in which
+split: 3 in this case, we insert in the *'From shuffle'* menu
```
Comment (n-poulsen) [inline guides/Benchmarking_shuffle_guide.md]:
```suggestion
`split: 3`. In this case, we insert in the *'From shuffle'* menu
```

File: guides/Benchmarking_shuffle_guide.md [hunk 3]
```diff
@@ -31,34 +43,57 @@ Creating a new shuffle with the same train/test split as an existing one:
...
-7. Click *'Create training dataset'* and move on to *'train network'*. Shuffle should be set to the new shuffle you entered at the previous step (in this case, 4)
+7. Click *'Create training dataset'* and move on to *'train network'*. Shuffle should be 
+set to the new shuffle you entered at the previous step (in this case, 4)
 ![create_from_existing](<assets/Screenshot 2024-07-29 at 17.47.10.png>)
-8. If you wish to keep the training attributes identical to your initial TensorFlow model, specifications of the original model can be found in the model folder dlc-models-pytorch > iteration folder (here 0) > shuffle (in this case 3) > train > pytorch_config.yaml. Here all parameters of the original model can be found.
+8. If you wish to keep the training attributes identical to your initial TensorFlow
+model, specifications of the original model can be found in the model folder
+dlc-models-pytorch > iteration folder (here 0) > shuffle (in this case 3) > train > 
+pytorch_config.yaml. Here all parameters of the original model can be found.
+
```
Comment (n-poulsen) [inline guides/Benchmarking_shuffle_guide.md:79]: I would edit this a bit

For TensorFlow models, the specifications for the net type and model configuration can be found in `dlc-models > iteration folder (here 0) > shuffle (in this case 3) > train > pose_config.yaml` (it's for PyTorch models that it's found in `pytorch_config.yaml`)

> If you wish to keep the training attributes identical to your initial TensorFlow

This isn't really possible: the only thing you can keep "identical" is the net type (such as using a ResNet 50). Otherwise the image augmentation, ... changed between PyTorch and TensorFlow, so really the net type is the only thing you can keep.

File: guides/Benchmarking_shuffle_guide.md [hunk 4]
```diff
@@ -31,34 +43,57 @@ Creating a new shuffle with the same train/test split as an existing one:
...
 #### In Code 
 
-With the *deeplabcut* module in Python, use the *create_training_dataset_from_existing_split()* method to create new shuffles from existing ones (e.g. TensorFlow shuffles)
+With the *deeplabcut* module in Python, use the
```
Comment (n-poulsen) [inline guides/Benchmarking_shuffle_guide.md]: This renders better in markdown with "``" symbols, which is "inline-code"

```suggestion
With the `deeplabcut` module in Python, use the
```

File: guides/Benchmarking_shuffle_guide.md [hunk 5]
```diff
@@ -31,34 +43,57 @@ Creating a new shuffle with the same train/test split as an existing one:
...
+*create_training_dataset_from_existing_split()* method to create new shuffles from
```
Comment (n-poulsen) [inline guides/Benchmarking_shuffle_guide.md]:
```suggestion
`create_training_dataset_from_existing_split()` method to create new shuffles from
```

File: guides/Benchmarking_shuffle_guide.md (hunk 1 above)
Comment (dikraMasrour) [reply guides/Benchmarking_shuffle_guide.md]: Here's a suggestion for this paragraph:
- When comparing different models, it's important to use the same train-test data split to ensure fair comparisons. If the models are trained on different datasets, their performance metrics can't be accurately compared. This is crucial when comparing the performance of models with different architectures or different sets of hyperparameters. For example, if we compare the RMSE of a model on an "easy" test image with the RMSE of another model on a "hard" test image, it doesn't determine whether a model is better than the other because the architecture performs better or because the training images were "better" to learn from. We need to compare the errors the models make on the same images by "decoupling" the dataset from the model architecture.

File: guides/Benchmarking_shuffle_guide.md (hunk 3 above)
Comment (dikraMasrour) [reply guides/Benchmarking_shuffle_guide.md:79]: How about this:

8. To view/edit the specifications of the model you created, you can go to `pytoch_config.yaml` file at:
```
dlc-project
|
|___ dlc-models-pytorch
     |__ iterationX
         |__ shuffleX
             |__ pytorch_config.yaml
```

Comment (n-poulsen) [reply guides/Benchmarking_shuffle_guide.md:79]: Yeah I think that would be great! I think we should update to use iterationX and shuffleX in the `Important files & folders` section too

File: guides/Benchmarking_shuffle_guide.md (hunk 1 above)
Comment (n-poulsen) [reply guides/Benchmarking_shuffle_guide.md]: I think that's very good - one thing is I would split

> For example, if we compare the RMSE of a model on an "easy" test image with the RMSE of another model on a "hard" test image, it doesn't determine whether a model is better than the other because the architecture performs better or because the training images were "better" to learn from.

into two sentences (one saying we need to compare metrics computed on the same test images, and one saying we need to train the models on the same images to "decouple" the dataset from the architecture)

File: guides/Benchmarking_shuffle_guide.md (hunk 3 above)
Comment (dikraMasrour) [reply guides/Benchmarking_shuffle_guide.md:79]: Right, just updated it

File: guides/Benchmarking_shuffle_guide.md (hunk 1 above)
Comment (dikraMasrour) [reply guides/Benchmarking_shuffle_guide.md]: How is this ?

- When comparing different models, it's important to use the same train-test data split to ensure fair comparisons. If the models are trained on different datasets, their performance metrics can't be accurately compared. This is crucial when comparing the performance of models with different architectures or different sets of hyperparameters. For example, if we compare the RMSE of a model on an "easy" test image with the RMSE of another model on a "hard" test image, it doesn't determine whether a model is better than the other because the architecture performs better or because the training images were "better" to learn from. Thus, we not only need to compare the models based on metrics computed on the same test images, but also train them on an identical fixed training set  in order to "decouple" the dataset from the model architecture.

File: guides/Benchmarking_shuffle_guide.md [hunk 6]
```diff
@@ -1,19 +1,32 @@
 # OUTLINE
```
Comment (dikraMasrour) [inline guides/Benchmarking_shuffle_guide.md]: Shall we think of a different title to the guide as well?

File: guides/Benchmarking_shuffle_guide.md (hunk 1 above)
Comment (n-poulsen) [reply guides/Benchmarking_shuffle_guide.md]: Good with me! Once this is updated, I think we can merge this into the main branch.

Then we can port the pull request to the main DeepLabCut repository

Comment (dikraMasrour) [reply guides/Benchmarking_shuffle_guide.md]: Yes, all changes have been added!
